### DISCUSSION TITLE
2D plot, prevent rotations

### DISCUSSION THREAD

**Discussion by nschloe** [pyvista/pyvista#2533] [Q&A] [answered]
For 2D meshes, I use
```python
p.view_xy()
```
to get a nice, flat representation of the data.

![screenshot](https://user-images.githubusercontent.com/181628/166106557-4aa40432-4645-45a0-a36f-70117592f329.png)

Clicking and dragging makes the plot behave like a 3D:

![screenshot](https://user-images.githubusercontent.com/181628/166106574-2d88a5d0-79ce-483d-ae2d-843031f98091.png)

Is there a way to prevent this? I just want x/y axes, not z, ever.

**Comment by tkoyama010** [comment, marked as answer]
~~It is my opinion that the current pyvista does not have the functionality for this, we need to add functionality to the `Plotter` class to be able to set `vtkInteractorStyleRubberBand2D`. Thank you for your request.~~
See https://github.com/pyvista/pyvista/discussions/2533#discussioncomment-2668844

**Reply by tkoyama010** [reply]
Ref: https://github.com/pyvista/pyvista/issues/2538

**Reply by nschloe** [reply]
Thanks for opening the feature request.

**Reply by MatthewFlamm** [reply]
Is this what you are referring to @tkoyama010 ?

https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_rubber_band_2d_style.html

**Reply by tkoyama010** [reply]
That is it! Thanks @MatthewFlamm and sorry to @nschloe my ignorance.

**Reply by MatthewFlamm** [reply]
Note that this style does not allow _any_ rotations, and it is unclear whether this is what you want.  The `enable_image_style` is similar, but allows rotations only around the viewing direction. It also allows the user to switch to different orthogonal views by hitting `x`, `y` or `z` keys, which probably isn't desired for your ask.  There is no exposed machinery to set or constrain the view direction yet in that interactor style.
